budgetportal/import_export_admin.py

```diff
@@ -0,0 +1,148 @@
+    """
+    Class to find a Department model instance from a row.
+    """
+
+    def get_instance(self, row):
+        """
+        Gets a Department instance by either a unique government-slug or
+        government-name combination, or if is_vote_primary is True, from a
+        unique government-vote_number combination.
+        """
+        name = self.resource.fields['name'].clean(row)
+        slug = slugify(name)
+        government = self.resource.fields['government'].clean(row)
+
+        instance = None
+
+        try:
+            instance = Department.objects.get(name=name, government=government)
+        except Department.DoesNotExist:
+            pass
+
+        if not instance:
+            try:
+                instance = Department.objects.get(slug=slug, government=government)
+            except Department.DoesNotExist:
+                pass
+
+        # If is_vote_primary, then (government, vote_number) must be unique
+        is_vote_primary = self.resource.fields['is_vote_primary'].clean(row)
+        vote_number = self.resource.fields['vote_number'].clean(row)
+        if is_vote_primary:
+            try:
+                instance = Department.objects.get(
+                    vote_number=vote_number, government=government)
+            except Department.DoesNotExist:
+                pass
+
+        if instance:
+            return instance
+
+        return None
+
+
+class DepartmentResource(resources.ModelResource):
+    """
+    Class to help django-import-export know how to map the rows in department
+    import files to django models.
+    """
+    is_vote_primary = Field(attribute='is_vote_primary',
+                            column_name='is_vote_primary', widget=CustomIsVotePrimaryWidget())
+    name = Field(attribute='name', column_name='department_name')
+    vote_number = Field(attribute='vote_number', column_name='vote_number')
+    government = Field(attribute='government',
+                       column_name='government', widget=CustomGovernmentWidget())
+
+    class Meta:
+        model = Department
+        fields = ('government', 'name', 'vote_number',
+                  'is_vote_primary', 'intro', 'website_url')
+        instance_loader_class = DepartmentInstanceLoader
+        import_id_fields = ['government', 'name']
+
+    def __init__(self, *args, **kwargs):
+        if 'sphere' in kwargs:
+            self.sphere = kwargs['sphere']
+            self.fields['government'].widget.set_sphere(self.sphere)
+
+
+class DepartmentImportForm(ImportForm):
+    """
+    Form class to use to upload a CSV file to import departments.
+    """
+    sphere = forms.ModelChoiceField(
+        queryset=Sphere.objects.all(),
+        required=True
+    )
+
+
+class CustomCSV(base_formats.TextFormat):
+    """
+    Class that will be used in the django-import-export module
+    to import data from a csv file. We override the create_dataset method
+    because the package's create_dataset method doesn't seem to work.
```

Comment:
This is due to https://github.com/django-import-export/django-import-export/issues/957 (tablib 0.13+ breaks django-import-export 1.2.0)

We can avoid the need for this workaround by keeping tablib to 0.12, which should be simpler and better overall. (The next release of django-import-export should fix this, but will also officially drop Python 2 support.)

Reply:
c81e1da should address this.